app/assets/javascripts/validation.js

```diff
@@ -154,6 +152,12 @@ function validateUsername(obj) {
 
 function validateEmail(obj) {
   var email = this.value;
+
+  if (email === "") {
```

Comment:
Here instead of checking if the `string === ""`, we can use regex to test if the string has any non-whitespace characters. That way the message will be displayed even if the email field contains  `"   "`. You can write it like this:
`if (/\S/.test(email)) {`

Reply:
Thanks, @Uzay-G. I love regexes also, but for someone new, it is not readable.
I can trim email and check for length. Much more readable and it has the same effect.